dirty_cat/_check_dependencies.py

```diff
@@ -0,0 +1,15 @@
+import pkg_resources
+
+
+def check_dependencies():
+    package_name = "dirty-cat"
+    env = pkg_resources.Environment()
+    package = env[package_name][0]
+    dependencies = package.requires()
+    for dep in dependencies:
+        installed_dep = env[dep.name][0]
+        if installed_dep not in dep:
+            raise ImportError(
+                f"{package_name} {package.version} requires {dep} "
+                f"but you have {installed_dep}, which is not compatible."
+            )
```

Comment:
Few comments from our discussion earlier:

- replace `env[package_name][0]` with something like `next((installed_dep for installed_dep in env[dep.name] if installed_dep.project_name == dep.name))` else you might have issues if there are several packages with names that contain the string `dep.name`

- I think it's a good idea to keep the `setup.py` with just an empty `setup()` instruction, so you can keep using things like `python setup.py bdist_wheel`, I don't think there's a replacement for this syntax yet.

- Feature-wise, we check that the direct requirements of `dirty-cat` are met but we don't check the whole dependency tree, there still might be issues at lower level of the dependency tree. (I think it is the right thing to do but I wanted to mention it clearly somewhere)

And finally maybe ensure that it is safe to use `pkg_resources` (for me I can't think of environments where it wouldn't be available or wouldn't be working correctly)